cmd/bosun/sched/sched.go

```diff
@@ -496,31 +507,33 @@ func marshalTime(t time.Time) string {
 var DefaultSched = &Schedule{}
 
 // Load loads a configuration into the default schedule.
-func Load(c *conf.Conf) error {
-	return DefaultSched.Load(c)
+func Load(systemConf conf.SystemConfProvider, ruleConf conf.RuleConfProvider, skipLast, quiet bool) error {
+	return DefaultSched.Load(systemConf, ruleConf, skipLast, quiet)
 }
 
 // Run runs the default schedule.
 func Run() error {
 	return DefaultSched.Run()
 }
 
-func (s *Schedule) Load(c *conf.Conf) error {
-	if err := s.Init(c); err != nil {
+func (s *Schedule) Load(systemConf conf.SystemConfProvider, ruleConf conf.RuleConfProvider, skipLast, quiet bool) error {
+	if err := s.Init(systemConf, ruleConf, skipLast, quiet); err != nil {
 		return err
 	}
 	if s.db == nil {
 		return nil
 	}
-	return s.RestoreState()
+	return nil
 }
 
-func Close() {
-	DefaultSched.Close()
+func Close(reload bool) {
+	DefaultSched.Close(reload)
 }
 
-func (s *Schedule) Close() {
-	if s.Conf.SkipLast {
+func (s *Schedule) Close(reload bool) {
+	s.cancelChecks()
+	s.checksRunning.Wait()
```

Comment:
The expressions are cancelled while running. So the only thing that we wait for to complete is the run history.